test/ngResource/resourceSpec.js

```diff
@@ -1356,5 +1357,36 @@ describe('resource', function() {
       );
   });
 
+  it('should cancel the request if timeout promise is resolved', function() {
+    var canceler = $q.defer();
+
+    $httpBackend.when('GET', '/CreditCard').respond({data: '123'});
+
+    var CreditCard = $resource('/CreditCard', {}, {
+      query: {
+        method: 'GET',
+        timeout: canceler.promise
+      }
+    });
+
+    CreditCard.query();
+
+    canceler.resolve();
+    expect($httpBackend.flush).toThrow(new Error("No pending request to flush !"));
+
+    canceler = $q.defer();
+    CreditCard = $resource('/CreditCard', {}, {
+      query: {
+        method: 'GET',
+        timeout: canceler.promise
+      }
+    });
+
+    CreditCard.query();
+    expect($httpBackend.flush).not.toThrow(new Error("No pending request to flush !"));
```

Comment:
I would leave it at `not.toThrow()` (accounting for the unlikely situation where the error message changes and the behaviour breaks (simultaneously)).
It's also more concise and readable.

Also, remove the 2 empty lines below.